specification/monitor/resource-manager/Microsoft.Insights/stable/2024-03-11/dataCollectionRules_API.json

```diff
@@ -0,0 +1,2379 @@
+          "example": [
+            "Microsoft-OtelLogs"
+          ]
+        },
+        "enrichWithResourceAttributes": {
+          "description": "Specifies the list of resource attributes that need to be added as labels/dimensions to the log record for further enrichment.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "example": [
+            "service.name"
+          ]
+        },
+        "enrichWithReference": {
+          "description": "Specifies the reference to enrich the telemetry signal with.",
+          "type": "string",
+          "example": "myAI"
+        },
+        "replaceResourceIdWithReference": {
+          "description": "Specifies whether to replace the default resourceId in the log record with the resourceId of the referenced resource being used for enrichment.",
+          "type": "boolean",
+          "example": true
+        },
+        "name": {
+          "description": "A friendly name for the data source.\r\nThis name should be unique across all data sources (regardless of type) within the data collection rule.",
+          "type": "string",
+          "example": "myDataSource1"
+        }
+      }
+    },
+    "OtelMetricsDataSource": {
+      "description": "Definition of OTel metrics configuration.",
+      "required": [
+        "streams"
+      ],
+      "type": "object",
+      "properties": {
+        "streams": {
+          "description": "List of streams that this data source will be sent to.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "example": [
+            "Custom-Metrics-MyMetrics"
+          ]
+        },
+        "resourceAttributeRouting": {
+          "description": "Specifies the routing policy based on OTLP payload resource attributes to route subset of the payload according to matching resource attribute.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/OtelDataSourceResourceAttributeRouting"
+            }
+          ]
+        },
+        "enrichWithResourceAttributes": {
+          "description": "Specifies the list of resource attributes that need to be added as labels/dimensions to the metric for further metric enrichment.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "example": [
+            "service.name",
+            "service.version"
+          ]
+        },
+        "enrichWithReference": {
+          "description": "Specifies the reference to enrich the telemetry signal with.",
+          "type": "string",
+          "example": "myAI"
+        },
+        "name": {
+          "description": "A friendly name for the data source.\r\nThis name should be unique across all data sources (regardless of type) within the data collection rule.",
+          "type": "string",
+          "example": "myDataSource1"
+        }
+      }
+    },
+    "OtelMetricsDirectDataSource": {
```

Comment:
>OtelMetricsDirectDataSource [](http://example.com/codeflow?start=5&length=27)

Earlier comments on similarly/identically-named properties apply to this definition as well.